htlcnotifier/interface.go

```diff
@@ -0,0 +1,21 @@
+package htlcnotifier
+
+// Notifier represents a place through which HTLC events can be piped through.
+//
+// Concrete implementations of Notifier should be non-blocking.
+type Notifier interface {
+	// NotifyForwardEvent is called when we successfully forward a HTLC.
+	NotifyForwardingEvent(f ForwardingEvent)
+
+	// NotifyLinkFailEvent is called when we fail to forward a HTLC over our
+	// outgoing link.
+	NotifyLinkFailEvent(reason FailReason, f ForwardingEvent)
+
+	// NotifySettleEvent is called when a HTLC we previously successfully
+	// forwarded is settled.
+	NotifySettleEvent(f ForwardingEvent)
```

Comment:
Probably need some way to correlate the forwarding event and the outcome (fail/settle). Payment hash is not enough, because it may be reused.

Reply:
Looking at using circuit `inKey`/`outKey` to identify HTLCs?
Forward events can have the `outKey` of the outgoing HTLC and they can be matched to settle/fail events.